Homogenizing geometry API (#2125)
ansys/pymapdl: doc/source/user_guide/mesh_geometry.rst

      Number of Element Components: 0

Geometry
--------

> Starting from PyMAPDL v0.66.0, by default, all the geometry entities are returned
> as a |pv-multiblock|  object. For instance:

Reviewer: ```suggestion
In PyMAPDL 0.66.0 and later, by default, all geometry entities are returned
as a |pv-multiblock|  object. For example:
```

.. code:: pycon

   >>> mapdl.geometry.areas
   MultiBlock (0x147ca7640)
     N Blocks    28
     X Bounds    -0.016, 0.016
     Y Bounds    -0.008, 0.018
     Z Bounds    -0.003, 0.015

> Or for example:

Reviewer: ```suggestion
**Example 2**:
```

.. code:: pycon

   >>> mapdl.geometry.keypoints
   MultiBlock (0x147a78220)
     N Blocks    26
     X Bounds    -0.016, 0.016
     Y Bounds    -0.008, 0.018
     Z Bounds    -0.003, 0.015

> As you can see, in the new API it is not needed to call the entities.

Reviewer: ```suggestion
As you can see, you do not need to call the entitities in the new API.
```

For more differences between the new Geometry API and the old one, see
:ref:`migration_geometry_api_066`. 

> The selection now is easier, allowing you to use indexing:

Reviewer: ```suggestion
The selection now is easier.

You can use indexing:
```

.. code:: pycon

   >>> volume0 = mapdl.geometry.volumes[0]
   >>> volume0
   UnstructuredGrid (0x149107340)
     N Cells:    34
     N Points:   36
     X Bounds:   0.000e+00, 1.588e-02
     Y Bounds:   -7.620e-03, 1.778e-02
     Z Bounds:   -3.180e-03, 0.000e+00
     N Arrays:   3

> Or using the entity name:

Reviewer: ```suggestion
Or, you can use the entity name:
```

.. code:: pycon

   >>> volume1 = mapdl.geometry.volumes["volume 1"]
   >>> volume1
   UnstructuredGrid (0x149107340)
     N Cells:    34
     N Points:   36
     X Bounds:   0.000e+00, 1.588e-02
     Y Bounds:   -7.620e-03, 1.778e-02
     Z Bounds:   -3.180e-03, 0.000e+00
     N Arrays:   3

> You can now plot the different entities by just issuing plot after them:

Author: No no... you got it right :). Editing...

> You can now plot the different entities by just issuing plot after them:

Author: ```suggestion
You can plot different entities by calling the :meth:`plot() <ansys.mapdl.core.mapdl_geometry.Geometry.keypoints.plot>` method:
```

.. code:: pycon

   >>> mapdl.geometry.areas.plot()

.. figure:: ../images/multiblock_pic1.png

> Or you can just plot a single entity:

Reviewer: ```suggestion
Or, you can plot a single entity:
```

| |get_lines|            | |pv-polydata|            | |list-pv-polydata|            | **Not Applicable**              | **Not Applicable**                  |
+------------------------+--------------------------+-------------------------------+---------------------------------+-------------------------------------+
| |get_areas|            | |pv-polydata|            | |list-pv-unstruc|             | **Not Applicable**              | **Not Applicable**                  |
+------------------------+--------------------------+-------------------------------+---------------------------------+-------------------------------------+
| |get_volumes|          | |pv-polydata|            | |list-pv-unstruc|             | **Not Applicable**              | **Not Applicable**                  |
+------------------------+--------------------------+-------------------------------+---------------------------------+-------------------------------------+

.. _migration_geometry_api_066:

Migration to the new Geometry API
---------------------------------

> Several changes need to be addressed in older to update your script
> from the old API to the new API.

Reviewer: ```suggestion
Several changes must be addressed in older scripts to update them
from the old API to the new API.
```

> One of the most important is that now, you don't need to call the entities like in the old API:

Reviewer: ```suggestion
One of the most important is that you no longer need to call the entities like you did in the old API:
```

.. code:: pycon

   # Old API
   >>> mapdl.geometry.areas()
   [UnstructuredGrid (0x7f14add95040)
     N Cells:	12
     N Points:	20
     X Bounds:	-2.000e+00, 2.000e+00
     Y Bounds:	0.000e+00, 1.974e+00
     Z Bounds:	0.000e+00, 0.000e+00
     N Arrays:	4,
   UnstructuredGrid (0x7f14add95ca0)
     N Cells:	12
     N Points:	20
     X Bounds:	-2.000e+00, 2.000e+00
     Y Bounds:	0.000e+00, 1.974e+00
     Z Bounds:	5.500e-01, 5.500e-01
     N Arrays:	4,
   ...

Whereas in the new API:

.. code:: pycon

   >>> mapdl.geometry.areas
   MultiBlock (0x147ca7640)
     N Blocks    28
     X Bounds    -0.016, 0.016
     Y Bounds    -0.008, 0.018
     Z Bounds    -0.003, 0.015

In addition, the type of entities returned by those methods are different.

> For comparison between the new and old geometry API, the following table is provided.

Reviewer: ```suggestion
This table compares the new and old APIs:
```

Author: I should have seen that one... 🤦

.. _Table-2:

> **Table 2. Comparison between objects returned by both geometry APIs.**

Reviewer: Should "geometry" here be capitalized? Is it referring to PyGeometry? I think within the topic, you should be consistent and "officially" define the "geometry API" in the first reference in the topic but then only use the "old API" and new "API". Sometimes throwing in "geometry" and sometimes leaving it out seems awkward.

Author: It is not referring to PyGeometry. I feel like having APIs "naked" is confusing. But yeah, I see your point.

> **Table 2. Comparison between objects returned by both geometry APIs.**

Author: ```suggestion
**Table 2. Comparison between objects returned by both APIs.**
```

+--------------------+------------------------------------------+--------------------------------------------------+
| **Function**       | **Old API**                              | **New API**                                      |

> |                    | (Function based - Needs to called)       | (Property based - Doesn't need to be called)     |

Reviewer: ```suggestion
|                    | (Function based - Must be called)        | (Property based - Doesn't need to be called)     |
```

+====================+==========================================+==================================================+
|  |geo-key|         | |np-array|                               | |pv-multiblock|                                  |
+--------------------+------------------------------------------+--------------------------------------------------+
| |geo-lines|        | |pv-polydata|                            | |pv-multiblock|                                  |
+--------------------+------------------------------------------+--------------------------------------------------+
| |geo-areas|        | |pv-polydata|                            | |pv-multiblock|                                  |
+--------------------+------------------------------------------+--------------------------------------------------+
| |geo-volumes|      | **Not existent**                         | |pv-multiblock|                                  |
+--------------------+------------------------------------------+--------------------------------------------------+

> For equivalence between the new and old geometry API, the following table is provided.

Reviewer: ```suggestion
This table shows the equivalence between the old and new APIs:
```

      Number of Element Components: 0

Geometry
--------

In PyMAPDL 0.66.0 and later, by default, all geometry entities are returned
as a |pv-multiblock|  object.

**Example 1**

.. code:: pycon

   >>> mapdl.geometry.areas
   MultiBlock (0x147ca7640)
     N Blocks    28
     X Bounds    -0.016, 0.016
     Y Bounds    -0.008, 0.018
     Z Bounds    -0.003, 0.015

**Example 2**:

.. code:: pycon

   >>> mapdl.geometry.keypoints
   MultiBlock (0x147a78220)
     N Blocks    26
     X Bounds    -0.016, 0.016
     Y Bounds    -0.008, 0.018
     Z Bounds    -0.003, 0.015

> As you can see, you do not need to call the entitities in the new API.

Author: ```suggestion
As you can see, you do not need to call the entities in the new API.
```
